api/konfig/builtinpluginconsts/namereference.go

```diff
@@ -140,6 +140,22 @@ nameReference:
     kind: ClusterRole
   - path: metadata/annotations/nginx.ingress.kubernetes.io\/fastcgi-params-configmap
     kind: Ingress
+  - path: spec/template/spec/containers/env/valueFrom/configMapKeyRef/name
+    kind: Service
+    group: serving.knative.dev
```

Comment:
@natasha41575 we've discussed that we only want to accept core types, but I see that we already have [one fieldspec](https://github.com/kubernetes-sigs/kustomize/blob/master/api/konfig/builtinpluginconsts/namereference.go#L290-L293) for knative, and that Jeff deliberately approved this: https://github.com/kubernetes-sigs/kustomize/issues/2564#issuecomment-644391655. Should we continue this exception?